physical/mssql/mssql.go

```diff
@@ -59,43 +51,50 @@ func NewMSSQLBackend(conf map[string]string, logger log.Logger) (physical.Backen
 		maxParInt = physical.DefaultParallelOperations
 	}
 
-	database, ok := conf["database"]
-	if !ok {
-		database = "Vault"
-	}
-
-	table, ok := conf["table"]
-	if !ok {
-		table = "Vault"
-	}
-
-	appname, ok := conf["appname"]
-	if !ok {
-		appname = "Vault"
+	// set defaults
+	defaults := map[string]string{
+		"database":           "Vault",
+		"table":              "Vault",
+		"appname":            "Vault",
+		"connection timeout": "30",
+		"schema":             "dbo",
+		"log level":          "0",
 	}
 
-	connectionTimeout, ok := conf["connectiontimeout"]
-	if !ok {
-		connectionTimeout = "30"
+	// backcomp maps commonly used hcl paramater names to valid sql server parameters for backwards compatibility
+	backcomp := map[string]string{
+		"connectiontimeout": "connection timeout",
+		"loglevel":          "log level",
+		"username":          "user id",
 	}
 
-	logLevel, ok := conf["loglevel"]
-	if !ok {
-		logLevel = "0"
-	}
-
-	schema, ok := conf["schema"]
-	if !ok || schema == "" {
-		schema = "dbo"
+	// upgrade parameter keys
+	var isSet bool
+	var confv string
+	for k, v := range backcomp {
+		confv, isSet = conf[k]
+		if isSet {
+			conf[v] = confv
+			delete(conf, k)
+		}
 	}
 
-	connectionString := fmt.Sprintf("server=%s;app name=%s;connection timeout=%s;log=%s", server, appname, connectionTimeout, logLevel)
-	if username != "" {
-		connectionString += ";user id=" + username
+	// inject defaults into configuration map
+	for k, v := range defaults {
+		_, isSet = conf[k]
+		if !isSet {
+			conf[k] = v
+		}
 	}
 
-	if password != "" {
-		connectionString += ";password=" + password
+	// create ado style connection unless a connection_url was provided
+	connectionString, ok := conf["connection_url"]
```

Comment:
Since the string zero value is `""`, you can simply do `connectionString := conf["connection_url"]` since you're already testing for `""` on the next line.